### PR TITLE
test dagster-dbt select with test tags

### PR DESCRIPTION
**Pull request by johannkm**
For https://github.com/dagster-io/dagster/issues/17372 - add coverage for the cases that work currently (deselecting tests, and selecting specific tests).

What doesn't work is selecting a test without a model. The error is bad- it falls back to everything. If we decide we'll keep this behavior, I'll fix up the error case

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17988** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17988?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17988?utm_source=stack-comment).

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml [hunk 1]
```diff
@@ -56,7 +57,8 @@ models:
         description: '{{ doc("orders_status") }}'
         tests:
           - accepted_values:
-              values: ['placed', 'shipped', 'completed', 'return_pending', 'returned']
```
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/models/schema.yml:59]: revert

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/partial_parse.msgpack
Comment (rexledesma) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/dbt_projects/test_dagster_asset_checks/partial_parse.msgpack:1]: remove these other artifacts (partial_parse, perf_info, semantic_manifest)

**Review by rexledesma — CHANGES_REQUESTED** [review]
Looks good, just remove the cruft

**Review by rexledesma — APPROVED** [review]

**Pull request merged**